packages/block-library/src/cover/edit.native.js

```diff
@@ -297,16 +332,98 @@ const Cover = ( {
 				/>
 			</PanelBody>
 
-			{ url ? (
-				<PanelBody title={ __( 'Media' ) }>
+			<PanelBody title={ __( 'Media' ) }>
+				{ url ? (
+					<>
+						<BottomSheet.Cell
+							style={ backgroundColor }
+							cellContainerStyle={ styles.mediaPreview }
+						>
+							<View style={ styles.mediaInner }>
+								{ IMAGE_BACKGROUND_TYPE === backgroundType && (
+									<Image
+										editButton={ true }
+										isSelected={ true }
+										isUploadFailed={ didUploadFail }
+										isUploadInProgress={
+											isUploadInProgress
+										}
+										onImageDataLoad={ () => {
+											setDisplayPlaceholder( false );
+										} }
+										onSelectMediaUploadOption={
+											onSelectMedia
+										}
+										openMediaOptions={
+											openMediaOptionsRef.current
+										}
+										url={ url }
+										height="100%"
+										style={ [
+											displayPlaceholder &&
+												styles.imagePlaceholder,
+										] }
+										width={ styles.image.width }
+									/>
+								) }
+								{ VIDEO_BACKGROUND_TYPE === backgroundType && (
+									<Video
+										muted
+										disableFocus
+										repeat
+										resizeMode={ 'contain' }
+										source={ { uri: url } }
+										onLoad={ onVideoLoad }
+										onLoadStart={ onVideoLoadStart }
+										style={
+											// Hide Video component since it has black background while loading the source
+											{ opacity: isVideoLoading ? 0 : 1 }
+										}
+									/>
+								) }
```

Comment:
What do you think if we unify this? We have almost the same content within `renderBackground`, maybe a `renderMedia` that has the `Image` and `Video` component to be used in both places.

Reply:
@geriux it may be possible, but there is divergence in the presentation of the media that may make it difficult.

Namely, the existing `renderBackground` is rendering media that should fill its parent's dimensions. It renders media that is absolutely positioned and height/width 100% inside of a parent with a min-height applied to it. For the bottom sheet, because [we removed application of the Cover block minimum height](https://appsp2.wordpress.com/2020/10/04/cover-block-media-settings-questions/#comment-2640), I believe we need to let the media itself dictate the sizing of the element. So, the styling and dimension computation logic applied within `renderBackground` and my additions are different.

Hopefully that makes sense. Please do let me know if you have additional thoughts or believe I a missing something. In the meantime, I'll review the two pieces of code to see if I can identify a way to share the code.

Reply:
Another note, additional changes to the video rendering were pushed between the time of your original comment and my own. So, it's likely worth reviewing those changes as well.

Reply:
I'll check it out =)